src/MongoDB.Analyzer/Core/Poco/PocoAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using MongoDB.Analyzer.Core.Utilities;
+
+namespace MongoDB.Analyzer.Core.Poco;
+
+internal static class PocoAnalyzer
+{
+    public static bool AnalyzePoco(MongoAnalysisContext context)
+    {
+        var sw = Stopwatch.StartNew();
+        var stats = AnalysisStats.Empty;
+        ExpressionsAnalysis pocoAnalysis = null;
+
+        try
+        {
+            context.Logger.Log("Started JSON analysis");
+
+            pocoAnalysis = PocoExpressionProcessor.ProcessSemanticModel(context);
+            stats = ReportJsonOrInvalidExpressions(context, pocoAnalysis);
+
+            sw.Stop();
+            context.Logger.Log($"JSON analysis ended: with {stats.JsonCount} JSON translations, {stats.DriverExceptionsCount} unsupported expressions, {stats.InternalExceptionsCount} internal exceptions in {sw.ElapsedMilliseconds}.");
+        }
+        catch (Exception ex)
+        {
+            sw.Stop();
+            context.Logger.Log($"JSON analysis ended after {sw.ElapsedMilliseconds}ms with exception {ex}");
+        }
+
+        var telemetry = AnalysisUtilities.GetTelemetry(pocoAnalysis, stats, sw);
+        if (telemetry.ExpressionsFound > 0)
+        {
+            context.Telemetry.PocoAnalysisResult(AnalysisUtilities.GetTelemetry(pocoAnalysis, stats, sw));
+        }
+
+        return telemetry.ExpressionsFound > 0;
+    }
+
+    private static AnalysisStats ReportJsonOrInvalidExpressions(MongoAnalysisContext context, ExpressionsAnalysis pocoAnalysis)
+    {
+        var semanticContext = context.SemanticModelAnalysisContext;
+        if (pocoAnalysis.AnalysisNodeContexts.EmptyOrNull())
+        {
+            return AnalysisStats.Empty;
+        }
+
+        var compilationResult = AnalysisCodeGenerator.Compile(context, pocoAnalysis);
+        if (!compilationResult.Success)
+        {
+            return AnalysisStats.Empty;
+        }
+
+        var driverVersion = compilationResult.PocoTestCodeExecutor.DriverVersion;
+        var settings = context.Settings;
+        int jsonCount = 0, internalExceptionsCount = 0, driverExceptionsCount = 0;
+
+        foreach (var analysisContext in pocoAnalysis.AnalysisNodeContexts)
+        {
+            var jsonResult = compilationResult.PocoTestCodeExecutor.GenerateJson(analysisContext.EvaluationMethodName);
+            var locations = analysisContext.Node.Locations;
+
+            if (jsonResult.Json != null)
+            {
+                var json = jsonResult.Json;
+                var diagnosticDescriptor = PocoDiagnosticRules.DiagnosticRulePoco2Json;
+                var decoratedMessage = DecorateMessage(json, driverVersion, context.Settings);
```

Comment:
No need for `json` and `diagnosticDescriptor` variables.

Reply:
Done